metadata-ingestion/src/datahub/ingestion/source/unity/source.py

```diff
@@ -269,10 +232,11 @@ def build_service_principal_map(self) -> None:
     def process_metastores(self) -> Iterable[MetadataWorkUnit]:
         metastores: Dict[str, Metastore] = {}
         assigned_metastore = self.unity_catalog_api_proxy.assigned_metastore()
-        if assigned_metastore:
-            metastores[assigned_metastore.metastore_id] = assigned_metastore
-        for metastore in self.unity_catalog_api_proxy.metastores():
-            metastores[metastore.metastore_id] = metastore
+        metastores[assigned_metastore.metastore_id] = assigned_metastore
+
+        if self.config.ingest_all_metastores:
+            for metastore in self.unity_catalog_api_proxy.metastores():
```

Comment:
do we have good error messages if this call fails?

Reply:
If you don't have perms you get the following:

```
[2023-05-23 15:49:36,909] ERROR    {datahub.entrypoints:199} - Command failed: Only account admin can list metastores. Config: host=https://dbc-a2674c21-c551.cloud.databricks.com, token=***, auth_type=pat
Traceback (most recent call last):
    <... long stack trace>
    response = self._workspace_client.metastores.list()
  File "/Users/asikowitz/acryldata/datahub/metadata-ingestion/venv/lib/python3.10/site-packages/databricks/sdk/service/catalog.py", line 3110, in list
    json = self._api.do('GET', '/api/2.1/unity-catalog/metastores')
  File "/Users/asikowitz/acryldata/datahub/metadata-ingestion/venv/lib/python3.10/site-packages/databricks/sdk/core.py", line 783, in do
    raise self._make_nicer_error(status_code=response.status_code, **payload) from None
databricks.sdk.core.DatabricksError: Only account admin can list metastores. Config: host=https://dbc-a2674c21-c551.cloud.databricks.com, token=***, auth_type=pat

```
And ingestion completely errors out. Personally I'm fine with this as getting metastores is a non-starter, but I can add a try catch if that's preferred.